heatmap/src/heatmap.rs

```diff
@@ -228,40 +274,41 @@ impl Heatmap {
         self.summary.buckets()
     }
 
+    /// Returns the number of valid and active `Histogram` slices in the `Heatmap`
+    pub fn active_slices(&self) -> usize {
+        let instant = self.tick_at.load(Ordering::Relaxed);
+        let elapsed = instant.duration_since(self.tick_origin);
+        let total_ticks = (elapsed.as_nanos() / self.resolution.as_nanos()) as usize;
+        min(total_ticks, self.slices() - 1)
+    }
+
     /// Increment a time-value pair by a specified count
-    pub fn increment(&self, time: Instant, value: u64, count: u32) {
-        self.tick(time);
+    pub fn increment(&self, time: Instant, value: u64, count: u32) -> Result<(), Error> {
+        let (tick_at, mut idx, ntick) = self.tick(time);
 
-        let current = self.current.load(Ordering::Relaxed);
+        let behind = tick_at.duration_since(time);
 
-        // fast path when the time falls into the current time slice
-        if time >= self.slices[current].start.load(Ordering::Relaxed) {
-            let _ = self.summary.increment(value, count);
-            let _ = self.slices[current].histogram.increment(value, count);
-        }
+        // slow path when the time falls into the current time slice
```

Comment:
I don't think this comment is correct? It seems like this is when the instant does not fall into the current time slice.